python_modules/dagster/dagster/components/resolved/context.py

```diff
@@ -18,8 +18,17 @@
 T = TypeVar("T")
 
 
-def env_scope(key: str) -> Optional[str]:
-    return os.environ.get(key)
+class EnvScope:
```

Comment:
while we are here, thoughts on 
* apis for return `None` vs throw on missing 
* indexing ie `env['MY_KEY']` (potentially related to previous point)